### PR TITLE
Add error handling middleware

### PR DESCRIPTION
**Pull request by kahmali**
- Add rest-json-error-handler package for handling Connect errors

See the discussion here for more context on this PR: https://github.com/stubailo/meteor-rest/pull/40#discussion_r31317901

### CODE REVIEW

File: packages/rest-json-error-handler/json_error_handler.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Response looks like:
+ *   {
+ *     error: 'Error type',
+  *    reason: 'Cause of error'
+  *  }
+ *
+ * @middleware
+ */
+RestMiddleware.handleErrorAsJson = function (error, request, response, next) {
+  console.log('************ handleErrorAsJson ***************');
```
Comment (kahmali) [inline packages/rest-json-error-handler/json_error_handler.js]: Sorry, I'll remove this unnecessary log statement. Just had it there for some cheap debugging.

File: packages/rest-json-error-handler/json_error_handler.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  var body = JSON.stringify({ error: error.error, reason: error.reason });
+
+  response.statusCode = error.statusCode;
+  response.setHeader('Content-Type', 'application/json');
+  response.write(body);
+  response.end();
+  //JsonRoutes.sendResult(response, 400, { error: error.error, reason: error.reason });
+  //next();
```
Comment (kahmali) [inline packages/rest-json-error-handler/json_error_handler.js]: Remove this unnecessary call to `next()` (can't call after response is sent)

**Comment by aldeed** [comment]
Hey @kahmali I'm trying to reboot getting this stuff merged and 3.0 released. Do you want to rebase and finish cleanup on this or should I take it over?

**Comment by kahmali** [comment]
Hey Eric! 

Hope all is well. Glad to see you reviving this. I'm swamped right now, as I'm a few weeks out from an MVP launch, and I'm just trying to get all my ducks in a row with that. So, unfortunately, I won't be able to contribute much, if at all, on this for the next few weeks. 

I did go ahead and rebase onto the latest and clean up the few things I mentioned above. I got it situated with jshint, except for the one error I mentioned in the commit message. The other known issue mentioned in the commit still exists, and the error handling middleware doesn't actually _work_ until that gets resolved, so this isn't ready to be merged in yet. I'm sure you'll be able to take it from here and get it working.

Sorry I can't be more help with this at the moment. Restivus is now at a point where it can share middleware with simple:rest, so as soon as I have some time I hope to do my part to push both projects along.

**Pull request merged**